testsuite/MDAnalysisTests/auxiliary/test_edr.py

```diff
@@ -0,0 +1,441 @@
+    def test_add_single_term_custom_name_from_reader(self, ref_universe,
+                                                     reader):
+        ref_universe.trajectory.add_auxiliary({"temp": "Temperature"}, reader)
+        ref_dict = get_auxstep_data(0)
+        assert ref_universe.trajectory.ts.aux.temp == ref_dict["Temperature"]
+
+# TODO: Change order of aux_spec and auxdata for 3.0 release, cf. Issue #3811
+    def test_terms_update_on_iter(self, ref_universe, reader):
+        ref_universe.trajectory.add_auxiliary({"bond": "Bond",
+                                               "temp": "Temperature"},
+                                              reader)
+        ref_dict = get_auxstep_data(0)
+        assert ref_universe.trajectory.ts.aux.bond == ref_dict["Bond"]
+        assert ref_universe.trajectory.ts.aux.temp == ref_dict["Temperature"]
+        ref_dict = get_auxstep_data(1)
+        ref_universe.trajectory.next()
+        assert ref_universe.trajectory.ts.aux.bond == ref_dict["Bond"]
+        assert ref_universe.trajectory.ts.aux.temp == ref_dict["Temperature"]
+
+# TODO: Change order of aux_spec and auxdata for 3.0 release, cf. Issue #3811
+    def test_invalid_data_selector(self, ref, ref_universe):
+        with pytest.raises(KeyError, match="'Nonsense' is not a key"):
+            ref_universe.trajectory.add_auxiliary({"something": "Nonsense"},
+                                                  AUX_EDR)
+
+    def test_read_all_times(self, reader):
+        all_times_expected = np.array([0., 0.02, 0.04, 0.06])
+        assert np.all(all_times_expected == reader.read_all_times())
+
+    def test_get_data_from_string(self, ref, reader):
+        returned = reader.get_data("Bond")
+        assert isinstance(returned, dict)
+        assert_almost_equal(ref.times, returned["Time"])
+        assert_almost_equal(ref.bonds, returned["Bond"])
+
+    def test_get_data_from_list(self, ref, reader):
+        returned = reader.get_data(["Bond", "Angle"])
+        assert isinstance(returned, dict)
+        assert_almost_equal(ref.times, returned["Time"])
+        assert_almost_equal(ref.bonds, returned["Bond"])
+        assert_almost_equal(ref.angles, returned["Angle"])
+
+    def test_get_data_everything(self, ref, reader):
+        returned = reader.get_data()
+        returned_asterisk = reader.get_data()
+        assert returned.keys() == returned_asterisk.keys()
+        ref_terms = [key for key in get_auxstep_data(0).keys()]
+        assert ref_terms == reader.terms
+        assert_almost_equal(ref.bonds, returned["Bond"])
+
+    def test_get_data_invalid_selections(self, reader):
+        with pytest.raises(KeyError, match="data selector"):
+            reader.get_data(42)
+        with pytest.raises(KeyError, match="data selector"):
+            reader.get_data("Not a valid term")
+        with pytest.raises(KeyError, match="data selector"):
+            reader.get_data(["Bond", "Not a valid term"])
```

Comment:
Rather than three separate checks for raises, it would be better if you just parametrized this method to loop over the match & get_data inputs.